src/main/java/com/google/devtools/build/remote/client/GrpcRemoteCache.java

```diff
@@ -57,13 +68,88 @@ public static boolean isRemoteCacheOptions(RemoteOptions options) {
     return options.remoteCache != null;
   }
 
+  private ContentAddressableStorageBlockingStub casBlockingStub() {
+    return ContentAddressableStorageGrpc.newBlockingStub(channel)
+        .withInterceptors(TracingMetadataUtils.attachMetadataFromContextInterceptor())
+        .withCallCredentials(credentials)
+        .withDeadlineAfter(options.remoteTimeout, TimeUnit.SECONDS);
+  }
+
   private ByteStreamBlockingStub bsBlockingStub() {
     return ByteStreamGrpc.newBlockingStub(channel)
         .withInterceptors(TracingMetadataUtils.attachMetadataFromContextInterceptor())
         .withCallCredentials(credentials)
         .withDeadlineAfter(options.remoteTimeout, TimeUnit.SECONDS);
   }
 
+  /**
+   * Download a tree with the {@link Directory} given by digest as the root directory of the tree.
+   * This method first attempts to retrieve the {@link Tree} using the GetTree RPC. If this rpc is
+   * unimplemented on the server, this method falls back to the default implementation {@link
+   * AbstractRemoteActionCache#getTree(Digest)} provided by its parent class.
+   *
+   * @param rootDigest The digest of the root {@link Directory} of the tree
+   * @return A tree with the given directory as the root.
+   * @throws IOException in the case that retrieving the blobs required to reconstruct the tree
+   *     failed.
+   */
+  @Override
+  public Tree getTree(Digest rootDigest) throws IOException {
+    byte directoryBytes[];
+    Directory dir;
+    directoryBytes = downloadBlob(rootDigest);
+    dir = Directory.parseFrom(directoryBytes);
+    try {
+      return getTreeUsingRpc(dir, rootDigest);
+    } catch (StatusRuntimeException e) {
+      if (e.getStatus().getCode() == Code.UNIMPLEMENTED) {
+        // Fallback to generic getTree if getTree RPC is unimplemented.
```

Comment:
It's implemented. OK, not by the RemoteWorker, but it's on my TODO list. Anyway, RBE does do it.

Reply:
I removed it.